### ISSUE TITLE
%f in statusline gives inconsistent path format

### ISSUE THREAD

**Issue by john-cj** [vim/vim#8858]
* Vim 8.2.2824 is installed in `c:\prgfiles\vim` (I use it on Windows 7)
* When I start it, the output of `:pwd` is `C:\Users\John`

My test vimrc is:

```
set laststatus=2
set statusline=%f
```

Now, three tests:

1. If I start Vim, then select "File", "Open", and select a file from `C:\Users\John\Desktop`, the text in status line is `Desktop\test.txt`

2. If I start Vim and then type `:e c:\users\john\desktop\test.txt`, the text in status line is `~\Desktop\test.txt`

3. If I start Vim, then select "File", "Open", and select a file from `C:\Foo`, the text in status line is `\Foo\Bar.txt`

As you see, there are different results in each test. Seems to be a bug, though I'm not really sure.

**Comment by brammool** [comment]
It may depend on your current directory, check the output of ":pwd".

**Comment by john-cj** [comment]
The output of `:pwd` is `C:\Users\John`.

**Comment by brammool** [comment]
Does the output of :ls  show the same as the %f item?

**Comment by john-cj** [comment]
Yes, in all three cases `:ls` shows the same as `%f`.

**Comment by brammool** [comment]
Well, then I think this is working as expected.  Maybe using ":cd ." updates the names.

**Comment by john-cj** [comment]
No, it doesn't. I'm not sure whether this issue should be closed.

On the one hand, I dont't think that if a user have opened mutiple files by using differnt ways (menu and `:e`), it is OK that file paths are shown in different formats.

On the other hand, you say it is technically correct (if I understood you correctly), and also it seems I'm the first user who is so picky.

**Comment by john-cj** [comment]
**Some more tests**

`:pwd` shows `C:\Users\John`

| Tasks                                                            | `%f`              | `%{expand('%:~:.')}`
| ---                                                              | ---               | ---
| To open `C:\Users\John\Desktop\x.txt` by using the menu          | `Desktop\x.txt`   | `Desktop\x.txt`
| To open the same file by typing `:e c:/users/john/desktop/x.txt` | `~\Desktop\x.txt` | `Desktop\x.txt`
| To open `C:\foo\bar\x.txt` by using the menu                     | `\foo\bar\x.txt`  | `\foo\bar\x.txt`
